projects/js-packages/publicize-components/src/components/social-post-modal/preview-section.tsx

```diff
@@ -88,7 +88,11 @@ export function PreviewSection() {
 							// in that case we won't show the toggle
 							! shouldBeDisabled( tab ) ? (
 								<ToggleControl
-									label={ __( 'Connection enabled', 'jetpack' ) }
+									label={
+										tab.enabled
+											? __( 'Connection enabled', 'jetpack' )
+											: __( 'Connection disabled', 'jetpack' )
+									}
 									checked={ canBeTurnedOn( tab ) && tab.enabled }
```

Comment:
For broken connections, it will still say, "Connection enabled" even thought it will actually be disabled because of that `checked={ canBeTurnedOn( tab ) && tab.enabled }` below.

I think it will be better to do something like this

```jsx
const isEnabled = anBeTurnedOn( tab ) && tab.enabled

label={ isEnabled ? __( 'Connection enabled', 'jetpack' ) : __( 'Connection disabled', 'jetpack' ) }
checked={ isEnabled }
```

Reply:
Thanks Manzoor. Implemented in 08a03e4